cpp/include/replace.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2018-2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef REPLACE_HPP
+#define REPLACE_HPP
+
+#include <cudf.h>
+#include <types.hpp>
+
+// Forward declaration
+typedef struct CUstream_st* cudaStream_t;
+
+namespace cudf {
+
+/**
+ * @brief Replaces all null values in a column with corresponding values of another column
+ *
+ * The first column is expected to be a regular gdf_column. The second column
+ * must be of the same type and same size as the first.
+ *
+ * The function replaces all nulls of the first column with the
+ * corresponding elements of the second column
+ *
+ * @param[in] input A gdf_column containing null values
+ * @param[in] replacement A gdf_column whose values will replace null values in input
+ * @param[in] stream Optional stream in which to perform allocations
+ *
+ * @returns gdf_column Column with nulls replaced
+ */
+gdf_column replace_nulls(const gdf_column& input,
+                         const gdf_column& replacement,
+                         cudaStream_t stream = 0);
```

Comment:
We're not ready to expose streams in libcudf's external APIs.

What I would do is to take these existing APIs, put them in a `detail` name space. Then, in the `cudf` namespace, copy these same APIs (minus the stream), and just call the corresponding API in `detail`. That way, it will be easy to use the stream when we're ready to do so.